src/DotVVM.Framework.Tests.Common/ResourceManagement/ResourceRepositoryTests.cs

```diff
@@ -25,5 +25,199 @@ public void ResourceRepository_CyclicDependency_Throws()
                 manager.AddRequiredResource("one");
             });
         }
+
+        [TestMethod]
+        public void ResourceRepository_CyclicDependency_Structure_1()
+        {
+            Assert.ThrowsException<DotvvmResourceException>(() => {
+                var configuration = DotvvmConfiguration.CreateDefault();
+
+                configuration.Resources.Register("7", new NullResource { Dependencies = new[] { "6", "5" } });
+
+                configuration.Resources.Register("6", new NullResource { Dependencies = new[] { "4", "3" } });
+                configuration.Resources.Register("5", new NullResource { Dependencies = new[] { "4", "3" } });
+
+                configuration.Resources.Register("4", new NullResource { Dependencies = new[] { "2", "1" } });
+                configuration.Resources.Register("3", new NullResource { Dependencies = new[] { "1" } });
+
+                configuration.Resources.Register("2", new NullResource { Dependencies = new[] { "dotvvm" } });
+                configuration.Resources.Register("1", new NullResource { Dependencies = new[] { "dotvvm" } });
+
+                var manager = configuration.ServiceProvider.GetRequiredService<ResourceManager>();
+                manager.AddRequiredResource("7");
```

Comment:
Where is the cycle? Why should this test throw an exception?

Reply:
Here is no cycle. If this throws an exception then there is a bug in the detection of cycles.

Reply:
@OndrejSulc wrote:
> this specific test is not cyclic and should not throw error. I think you
copy pasted wrong one because there are two similar ones. This one you sent
me and other which has resource "dotvvm" dependent on resource "7" (makes it
cyclic). So you are looking for ResourceRepository_CyclicDependency_
Structure_2_Throws(). This is ResourceRepository_CyclicDependency_Structure_
1() and is not cyclic as you said correctly.

Well, you have the test inside the `Assert.ThrowsException<DotvvmResourceException>(() => { ... })` method that would fail the test if it did not throw. If you meant to test that it does not throw, you should remove the invocation of `Assert.ThrowsException`. The `_Throws` suffix in the name is ignored by testing framework, the names are just for humans.